### PR TITLE
Improved the explanation about the "secret" configuration parameter

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | all |
| Fixed tickets | - |

### CODE REVIEW

**Comment by dupuchba** [comment]
:+1:

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -60,10 +60,24 @@ secret
...
+to add more entropy to security related operations. Its value should be series of
+characters, numbers and symbols choosen randomly. It's recommended length is
+around 32 characters.
+
+In practice, Symfony uses this value for generating the :ref:`CSRF tokens <forms-csrf>`,
+for encrypting the cookies used in the :doc:`remember me functionality </cookbook/security/remember_me>`
+and for creating signed URIs when using :ref:`ESI (Edge Side Includes) <edge-side-includes>` .
+
+This option becomes the service container parameter named ``kernel.secret``,
+which you can use whenever the application needs a immutable random string
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: an immutable

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -60,10 +60,24 @@ secret
 
 **type**: ``string`` **required**
 
-This is a string that should be unique to your application. In practice,
-it's used for generating the CSRF tokens, but it could be used in any other
-context where having a unique string is useful. It becomes the service container
-parameter named ``kernel.secret``.
+This is a string that should be unique to your application and it's commonly used
+to add more entropy to security related operations. Its value should be series of
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: a series

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -60,10 +60,24 @@ secret
...
+
+In practice, Symfony uses this value for generating the :ref:`CSRF tokens <forms-csrf>`,
+for encrypting the cookies used in the :doc:`remember me functionality </cookbook/security/remember_me>`
+and for creating signed URIs when using :ref:`ESI (Edge Side Includes) <edge-side-includes>` .
+
+This option becomes the service container parameter named ``kernel.secret``,
+which you can use whenever the application needs a immutable random string
+to add more entropy.
+
+As any other security-related parameter, is a good practice to change this
```
Comment (xabbuh) [inline reference/configuration/framework.rst]: As with any [...]

File: reference/configuration/framework.rst [hunk 4]
```diff
@@ -60,10 +60,24 @@ secret
...
+
+In practice, Symfony uses this value for generating the :ref:`CSRF tokens <forms-csrf>`,
+for encrypting the cookies used in the :doc:`remember me functionality </cookbook/security/remember_me>`
+and for creating signed URIs when using :ref:`ESI (Edge Side Includes) <edge-side-includes>` .
+
+This option becomes the service container parameter named ``kernel.secret``,
+which you can use whenever the application needs an immutable random string
+to add more entropy.
+
+As with any other security-related parameter, is a good practice to change this
```
Comment (WouterJ) [inline reference/configuration/framework.rst]: "[...]parameter, it is a [...]"

Comment (javiereguiluz) [inline reference/configuration/framework.rst]: Fixed. Thanks.

**Comment by ricardclau** [comment]
:+1: to this. The sentence `delete the HTTP Cache related cache` sound a bit weird but I cannot think of a better way to say it in a generic way.

**Comment by javiereguiluz** [comment]
@ricardclau you are right about that sentence. It sounds awful. I've just reworded it. Thanks Ricard.

**Comment by weaverryan** [comment]
Wonderful explanation! Thanks Javier!

**Pull request merged**